packages/@aws-cdk/aws-servicediscovery/lib/service.ts

```diff
@@ -0,0 +1,448 @@
+}
+
+/**
+ * Settings for one DNS record that you want AWS Cloud Map to create when you register an instance.
+ */
+export interface DnsRecord {
+  /**
+   * The record type
+   */
+  type: DnsRecordType;
+
+  /**
+   * The time to live for the record
+   */
+  ttl: string;
+}
+
+/**
+ * Settings for an optional Amazon Route 53 health check. If you specify settings for a health check, AWS Cloud Map
+ * associates the health check with all the records that you specify in DnsConfig. Only valid with a PublicDnsNamespace.
+ */
+export interface HealthCheckConfig {
+  /**
+   * The type of health check that you want to create, which indicates how Route 53 determines whether an endpoint is
+   * healthy. Cannot be modified once created. Supported values are HTTP, HTTPS, and TCP.
+   *
+   * @default HTTP
+   */
+  type?: HealthCheckType;
+
+  /**
+   * The path that you want Route 53 to request when performing health checks. Do not use when health check type is TCP.
+   *
+   * @default '/'
+   */
+  resourcePath?: string;
+
+  /**
+   * The number of consecutive health checks that an endpoint must pass or fail for Route 53 to change the current
+   * status of the endpoint from unhealthy to healthy or vice versa.
+   *
+   * @default 1
+   */
+  failureThreshold?: number;
+}
+
+/**
+ * Specifies information about an optional custom health check.
+ */
+export interface HealthCheckCustomConfig {
+  /**
+   * The number of 30-second intervals that you want Cloud Map to wait after receiving an
+   * UpdateInstanceCustomHealthStatus request before it changes the health status of a service instance.
+   *
+   * @default 1
+   */
+  failureThreshold?: number;
+}
+
+export enum DnsRecordType {
+  /**
+   * An A record
+   */
+  A = "A",
+
+  /**
+   * An AAAA record
+   */
+  AAAA = "AAAA",
+
+  /**
+   * Both an A and AAAA record
+   */
+  A_AAAA = "A, AAAA",
+
+  /**
+   * A Srv record
+   */
+  Srv = "SRV",
+
+  /**
+   * A CNAME record
+   */
+  Cname = "CNAME",
+}
+
+export enum RoutingPolicy {
+  /**
+   * Route 53 returns the applicable value from one randomly selected instance from among the instances that you
+   * registered using the same service.
+   */
+  Weighted = "WEIGHTED",
+
+  /**
+   * If you define a health check for the service and the health check is healthy, Route 53 returns the applicable value
+   * for up to eight instances.
+   */
+  Multivalue = "MULTIVALUE",
+}
+
+export enum HealthCheckType {
+  /**
+   * Route 53 tries to establish a TCP connection. If successful, Route 53 submits an HTTP request and waits for an HTTP
+   * status code of 200 or greater and less than 400.
+   */
+  Http = "HTTP",
+
+  /**
+   * Route 53 tries to establish a TCP connection. If successful, Route 53 submits an HTTPS request and waits for an
+   * HTTP status code of 200 or greater and less than 400.  If you specify HTTPS for the value of Type, the endpoint
+   * must support TLS v1.0 or later.
+   */
+  Https = "HTTPS",
```

Comment:
I kind of cringe when I see `HTTP[S]` pascal-cased like that. But maybe it's just me 🤷🏻‍♂️